app/store/sagas/clubs.js

```diff
@@ -8,44 +8,41 @@ import {
 import { updateEntity, setEntity } from "../actions/entities";
 import { setCurrentClubID } from "../actions/data";
 
-import { token } from "../state/appState/selectors";
+import { token, getCurrentUserID } from "../state/appState/selectors";
 
-import {
-  listClubMembers,
-  listClubAnnouncements,
-  listClubActivities
-} from "../services/clubs";
+import { listClubMembers, listClubActivities } from "../services/clubs";
 
 import { getStats } from "./athlete";
 import { getRankings } from "../services/activities";
 
 function* listMembers() {
   const accessToken = yield select(token);
+  const currentUserID = yield select(getCurrentUserID);
   const clubID = 288750;
   yield put(setCurrentClubID("loading"));
   const { ids, error } = yield call(listClubMembers, accessToken, clubID);
   if (!error) {
     yield put(setCurrentClubID(clubID));
     yield put(updateEntity(clubID, "clubs", { members: ids }));
-    for (let index = 0; index < ids.length; index += 1) {
-      yield getStats(ids[index]);
+    const filteredIds = ids.filter(id => id !== currentUserID);
+    for (let index = 0; index < filteredIds.length; index += 1) {
+      yield getStats(filteredIds[index]);
```

Comment:
we should be more carefull here with sagas.
may be getStats should return the stats or may be we should buffer the stats
anyway i don't think we should trigger multiple puts like that it will trigger render each time and more